conda-store-server/conda_store_server/app.py

```diff
@@ -663,15 +673,23 @@ def register_environment(
             description=specification.spec["description"],
         )
 
-        build = self.create_build(db, environment.id, specification.sha256)
+        build = self.create_build(
+            db, environment.id, specification.sha256, is_lockfile=is_lockfile
+        )
 
         if environment_was_empty:
             environment.current_build = build
             db.commit()
 
         return build.id
 
-    def create_build(self, db: Session, environment_id: int, specification_sha256: str):
+    def create_build(
+        self,
+        db: Session,
+        environment_id: int,
+        specification_sha256: str,
+        is_lockfile: bool = False,
+    ):
```

Comment:
```suggestion
    def create_build(self, db: Session, environment_id: int, specification_sha256: str):
```

Revert this back, `is_lockfile` is not used later (it's read from the DB in that function).